### ISSUE TITLE
isatab.py Handle multiple columns with the same names

### ISSUE THREAD

**Issue by hcwi** [ISA-tools/isa-api#274]
Allow the same headers to occur multiple times in columns of the TAB file. Exemplary usage:
- Characteristics[someID] for Source and Sample,
- Parameter Value[treatmentType] for Protocol of type Treatment (could be used multiple times),
- Comment[rev20171115] assigned to different entities

Currently, in the process of reading the file with duplicated column names, the headers get indexed by  a suffix e.g. `Comment[rev20171115]`, `Comment[rev20171115].1`, `Comment[rev20171115].2` which isn't handled in further processing, resulting in the fragment `].` staying in the data and its json representation. 
Perhaps you could replace the current approach (as below) with a regex to handle this?
https://github.com/ISA-tools/isa-api/blob/49c36f8d180057832a1ab0711dacff07e3d04926/isatools/isatab.py#L3533
https://github.com/ISA-tools/isa-api/blob/49c36f8d180057832a1ab0711dacff07e3d04926/isatools/isatab.py#L3559
https://github.com/ISA-tools/isa-api/blob/49c36f8d180057832a1ab0711dacff07e3d04926/isatools/isatab.py#L3398

**Comment by hcwi** [comment]
Follow-up: The isa-api validator allows to have two identical columns.

What's more, it doesn't seem to take into account the position of the attributes, so that two columns with same headers are not distinguished. As a result, a tabular file `SourceName | Sample Name | Characteristics[x]` satisfies the validation against a config requiring `x` for both Source and Sample nodes.

**Comment by hcwi** [comment]
Yes, I confirm. 
The field is required and declared like below for both Source and Sample :

<field header="Source Name" data-type="String" is-file-field="false" is-multiple-value="false" is-required="true" is-hidden="false" is-forced-ontology="false"> ... </field>		
<field header="Characteristics[x]" data-type="Ontology term" is-file-field="false" is-multiple-value="false" is-required="true" is-hidden="false" is-forced-ontology="false">... </field>
...
<field header="Sample Name" data-type="String" is-file-field="false" is-multiple-value="false" is-required="true" is-hidden="false" is-forced-ontology="false">...</field>
<field header="Characteristics[x]" data-type="Ontology term" is-file-field="false" is-multiple-value="false" is-required="true" is-hidden="false" is-forced-ontology="false">... </field>

**Comment by djcomlab** [comment]
OK thanks, I'll look into it...

**Comment by djcomlab** [comment]
Fixed in 0943ef8e7a71d2a70b54bd1b956e059a7c6ddefc

**Comment by djcomlab** [comment]
Reopening as this doesn't quite work as expected, it seems to output like

```
Comment[<_sre.SRE_Match object; span=(0, 21), match='Comment[SRA Lab Name]'>]	Oxford e-Research Centre
```

so not getting the content out of the regex correctly.

**Comment by djcomlab** [comment]
Re-fixed in 66cb2923908545cfa97511a030dafb36f27d1550
